app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -0,0 +1,448 @@
+import android.view.animation.DecelerateInterpolator
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.doOnPreDraw
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import com.takusemba.spotlight.OnSpotlightListener
+import com.takusemba.spotlight.OnTargetListener
+import com.takusemba.spotlight.Spotlight
+import com.takusemba.spotlight.Target
+import com.takusemba.spotlight.shape.Circle
+import com.takusemba.spotlight.shape.RoundedRectangle
+import com.takusemba.spotlight.shape.Shape
+import org.oppia.android.R
+import org.oppia.android.app.fragment.FragmentComponentImpl
+import org.oppia.android.app.fragment.InjectableFragment
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.app.model.SpotlightViewState
+import org.oppia.android.app.onboarding.SpotlightNavigationListener
+import org.oppia.android.app.topic.PROFILE_ID_ARGUMENT_KEY
+import org.oppia.android.databinding.BottomLeftOverlayBinding
+import org.oppia.android.databinding.BottomRightOverlayBinding
+import org.oppia.android.databinding.TopLeftOverlayBinding
+import org.oppia.android.databinding.TopRightOverlayBinding
+import org.oppia.android.domain.spotlight.SpotlightStateController
+import org.oppia.android.util.accessibility.AccessibilityServiceImpl
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import java.util.LinkedList
+import java.util.Locale
+import javax.inject.Inject
+
+class SpotlightFragment : InjectableFragment(), SpotlightNavigationListener {
+  @Inject
+  lateinit var activity: AppCompatActivity
+
+  @Inject
+  lateinit var spotlightStateController: SpotlightStateController
+
+  @Inject
+  lateinit var accessibilityServiceImpl: AccessibilityServiceImpl
+
+  private var targetList = LinkedList<Target>()
+  private lateinit var spotlight: Spotlight
+  private var screenHeight: Int = 0
+  private var screenWidth: Int = 0
+  private lateinit var overlayBinding: Any
+  private var internalProfileId: Int = -1
+  private val isSpotlightActive = MutableLiveData(false)
+  private var isRTL = false
+
+  private fun calculateScreenSize() {
+    val displayMetrics = DisplayMetrics()
+    activity.windowManager.defaultDisplay.getMetrics(displayMetrics)
+
+    screenHeight = displayMetrics.heightPixels
+    screenWidth = displayMetrics.widthPixels
+  }
+
+  fun getSpotlightStatusLiveData(): MutableLiveData<Boolean> {
+    return isSpotlightActive
+  }
+
+  // since this fragment does not have any view to inflate yet, all the tasks should be done here.
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    (fragmentComponent as FragmentComponentImpl).inject(this)
+
+    internalProfileId = arguments?.getInt(PROFILE_ID_ARGUMENT_KEY) ?: -1
+    calculateScreenSize()
+    checkIsRTL()
+  }
+
+  fun requestSpotlightViewWithDelayedLayout(spotlightTarget: SpotlightTarget) {
+    spotlightTarget.anchor.doOnPreDraw {
```

Comment:
This still needs justification for why it's correct.

Reply:
added kdocs